tests/metrics.test.ts

```diff
@@ -284,26 +286,83 @@ describe('getMoveDistanceSingleGame', () => {
       moves: game,
     });
 
-    expect(result.maxDistancePiece).toBe('pe');
+    expect(result.maxDistancePiece).toBe('PE');
     expect(result.maxDistance).toEqual(2);
   });
 });
 
 // game being tested: https://www.chess.com/analysis/game/pgn/4uURW4rJaa?tab=analysis
-// describe.only('getKillDeathRatios', () => {
-//   it('should return the correct number of kills, deaths, and assists for each piece in a game', async () => {
-//     const game = [
-//       {
-//         metadata: [],
-//         moves: '1. e4 e5 2. d4 exd4 3. Qxd4 Nc6 4. Qa4 Nf6 5. Nc3 d5 6. exd5 Qe7+ 7. Kd1 Bg4+ 8. Kd2 Nxd5 9. Nb5 Ncb4 10. c3 O-O-O 11. f3 Qe3+ 12. Kd1 Nxc3# 0-1'
-//       }
-//     ];
-
-//     const result = await getkillDeathRatios(game);
-
-//     console.log(result);
-
-//     expect(result['PE'].kills).toEqual(1);
-//     expect(result['PE'].deaths).toEqual(1);
-//   });
-// })
+describe('getKillDeathRatios', () => {
+  it('should return the correct number of kills, deaths, and assists for each piece in a game', async () => {
+    const game = [
+      {
+        metadata: [],
+        moves: '1. e4 e5 2. d4 exd4 3. Qxd4 Nc6 4. Qa4 Nf6 5. Nc3 d5 6. exd5 Qe7+ 7. Kd1 Bg4+ 8. Kd2 Nxd5 9. Nb5 Ncb4 10. c3 O-O-O 11. f3 Qe3+ 12. Kd1 Nxc3# 0-1'
+      }
+    ];
+
+    const result = await getKillDeathRatios(game);
+
+    console.log('result', result);
+
+    expect(result.killsDeathsAssistsMap['pe'].kills).toEqual(1);
+    expect(result.killsDeathsAssistsMap['pe'].deaths).toEqual(1);
```

Comment:
can we add the Q & other pieces to this test too? Not a requirement but would be nice. And do you compute it by piece typse?

Reply:
can. computed by piece unambig piece type rn yes